kernel-spark/src/test/scala/io/delta/kernel/spark/snapshot/unitycatalog/UCManagedTableSnapshotManagerSuite.scala

```diff
@@ -0,0 +1,432 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.spark.snapshot.unitycatalog
+
+import java.util.Optional
+
+import io.delta.kernel.exceptions.KernelException
+import io.delta.kernel.spark.exception.VersionNotFoundException
+import io.delta.kernel.unitycatalog.{InMemoryUCClient, UCCatalogManagedClient, UCCatalogManagedTestUtils}
+import io.delta.storage.commit.uccommitcoordinator.InvalidTargetTableException
+
+import org.scalatest.funsuite.AnyFunSuite
+
+/** Integration tests for [[UCManagedTableSnapshotManager]]. */
+class UCManagedTableSnapshotManagerSuite
```

Comment:
@huan233usc what is our policy for all the tests in the v2 module? are they all supposed to be java or scala? whats the logic behind the policy?

Reply:
okay.. this sucks.